suite-native/device-authorization/src/hooks/useAuthorizationGoBack.ts

```diff
@@ -16,11 +16,14 @@ type NavigationProp = StackToTabCompositeProps<
 export const useAuthorizationGoBack = () => {
     const navigation = useNavigation<NavigationProp>();
 
+    // The authorization screens are in a stack navigator and when authorization is cancelled or succesfull, we want to
+    // go back to previous (non-authorization) stack. This is achieved by navigating to first screen of current stack
+    // and then going back to previous stack.
     const handleGoBack = () => {
-        // The authorization screens are in a stack navigator and when authorization is cancelled or succesfull, we want to
-        // go back to previous (non-authorization) stack. This is achieved by navigating to first screen of current stack
-        // and then going back to previous stack.
-        navigation.popToTop();
+        // If there is just one route in stack, popToTop is not needed and simple goBack is enough. Otherwise, popToTop will do goBack and it will be called twice.
```

Comment:
Comment is not clear to me, what will be called twice? Sounds like goBack is called twice, but I don't see it.

Reply:
I rephrased it, do you think it's better? https://github.com/trezor/trezor-suite/pull/13121/commits/6a65d5b34bc6e7dec59efe5253fdc2d2a5130550

Reply:
not really 🙉